### DISCUSSION TITLE
0068 XLS-68d: Sponsored Fees and Reserves

### DISCUSSION THREAD

**Discussion by mvadari** [XRPLF/XRPL-Standards#196] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by ObiajuluM** [comment]
Out of curiosity, wouldn't it be better if account x could submit a signed transaction on behalf account y so account y would bear the txn cost.

**Reply by wojake** [reply]
If I'm understanding your question correctly, you want Account X to sign a transaction using their own private key (correlating to Account X's public key) but Account Y would bear the transaction's cost and the transaction affects Account Y's holdings & settings?

You could already do that by delegating your account to another public key by adding them as a Regular Key or adding them on multi-sig list.

Otherwise, this proposal delivers that.

**Reply by mvadari** [reply]
This design was considered and discarded - see Appendix B.2. Both Account X and Account Y must agree to enter into this sponsorship relationship.

**Comment by tequdev** [comment]
Does the sponsored account itself not require 10XRP for account reserves?

If 10XRP is not required, then the new Account Reserve Calculation should subtract 1 acctReserve if the AccountRoot has `SponsorAccount`.

**Reply by mvadari** [reply]
Good catch - updated the equation.

**Reply by tequdev** [reply]
Perhaps `acct.SponsorAccount ? 0 : acctReserve` is correct.

**Reply by mvadari** [reply]
Oops - fixed

**Comment by tequdev** [comment]
Currently a PriceOracle object requires an incremental reserve of 1 or 2.

If an object with such a dynamic incremental reserve increases its reserve requirement from 1 to 2 (object update), is it necessary to specify a Sponsor field to the txn?

**Reply by mvadari** [reply]
I don't think it makes sense to allow the ability to split up the reserve - that adds a whole lot of extra complexity.

**Reply by tequdev** [reply]
Is it possible to update the PriceOracle object to increase the incremental reserve from 1 to 2 without explicitly specifying the Sponsor field?

**Reply by mvadari** [reply]
You mean for a sponsored object? I think you'd probably need another sponsor signature for that, otherwise you're essentially able to lock up their XRP without their permission.

**Comment by tequdev** [comment]
If a trust line is automatically created when an offer you created is executed, is it possible to have someone else sponsor it? How?

**Reply by mvadari** [reply]
Hmm probably not. But I'm open to ideas. I suppose theoretically there could be an additional field on `OfferCreate` for that.

**Reply by tequdev** [reply]
same for ttNFTokenAcceptOffer.

As a result of accepting a BuyOffer, a new NFTokenPage may be created for the Buyer, so we will need to think about sponsoring this as well.

**Reply by mvadari** [reply]
This feels like an addition that is not necessary in a V1 implementation of this feature.
